presto-geospatial-toolkit/src/main/java/com/facebook/presto/geospatial/JtsGeometrySerde.java

```diff
@@ -32,319 +32,206 @@
         }
     }
 
-    private static Point readPoint(SliceInput input)
+    private static Point readPoint(SliceInput input, boolean skipType)
     {
-        requireNonNull(input, "input is null");
+        if (skipType) {
+            skipEsriType(input);
+        }
         Coordinate coordinates = readCoordinate(input);
         if (isEsriNaN(coordinates.x) || isEsriNaN(coordinates.y)) {
             return GEOMETRY_FACTORY.createPoint();
         }
         return GEOMETRY_FACTORY.createPoint(coordinates);
     }
 
-    private static void writePoint(Point point, SliceOutput output, boolean inGeometryCollection)
-    {
-        requireNonNull(output, "output is null");
-        if (inGeometryCollection) {
-            output.writeInt(SIZE_OF_DOUBLE * 2 + SIZE_OF_INT);
-        }
-        else {
-            output.writeByte(GeometryType.POINT.code());
-        }
-
-        output.writeInt(EsriShapeType.POINT.code);
-        if (!point.isEmpty()) {
-            writeCoordinate(point.getCoordinate(), output);
-        }
-        else {
-            writeCoordinate(new Coordinate(Double.NaN, Double.NaN), output);
-        }
-    }
-
-    private static Coordinate readCoordinate(SliceInput input)
-    {
-        requireNonNull(input, "input is null");
-        return new Coordinate(input.readDouble(), input.readDouble());
-    }
-
-    private static void writeCoordinate(Coordinate coordinate, SliceOutput output)
-    {
-        requireNonNull(output, "output is null");
-        output.writeDouble(coordinate.x);
-        output.writeDouble(coordinate.y);
-    }
-
-    private static void writeCoordinates(Coordinate[] coordinates, SliceOutput output)
-    {
-        requireNonNull(coordinates, "coordinates is null");
-        requireNonNull(output, "output is null");
-        for (Coordinate coordinate : coordinates) {
-            writeCoordinate(coordinate, output);
-        }
-    }
-
     private static Geometry readMultiPoint(SliceInput input)
     {
-        requireNonNull(input, "input is null");
+        skipEsriType(input);
         skipEnvelope(input);
         int pointCount = input.readInt();
         Point[] points = new Point[pointCount];
         for (int i = 0; i < pointCount; i++) {
-            points[i] = readPoint(input);
+            points[i] = readPoint(input, false);
         }
         return GEOMETRY_FACTORY.createMultiPoint(points);
     }
 
-    private static void writeMultiPoint(MultiPoint geometry, SliceOutput output, boolean inGeometryCollection)
+    private static Geometry readPolyline(SliceInput input, boolean multitype)
     {
-        requireNonNull(geometry, "geometry is null");
-        requireNonNull(output, "output is null");
+        skipEsriType(input);
+        skipEnvelope(input);
+        int partCount = input.readInt();
+        if (partCount == 0) {
+            if (multitype) {
+                return GEOMETRY_FACTORY.createMultiLineString();
+            }
+            else {
```

Comment:
unnecessary `else`

Reply:
It was a typo. I should've return `GEOMETRY_FACTORY.createLineString()` in `else` branch.